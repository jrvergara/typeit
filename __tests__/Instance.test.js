import Instance from '../src/Instance.js';

let instance;

beforeEach(() => {
  document.body.innerHTML = `
    <div>
      <span id="element"></span>
    </div>
  `;

  instance = new Instance(
    document.getElementById('element'),
    'arbitrary-id',
    {},
    true,
    null
  )
});

describe("queueDeletions()", () => {
  test("Queues string length when simple string is passed.", () => {
    instance.queueDeletions('hello');
    expect(instance.queue).toHaveLength(6);
    expect(instance.queue).toMatchSnapshot();
  });

  test("Queues number when number is passed.", () => {
    instance.queueDeletions(6);
    expect(instance.queue).toHaveLength(7);
    expect(instance.queue).toMatchSnapshot();
  });

  test("Queues correct length when HTML is passed.", () => {
    instance.queueDeletions('Some <strong>HTML</strong>.');
    expect(instance.queue).toHaveLength(11);
    expect(instance.queue).toMatchSnapshot();
  });

  test("Queues correct length when multiple HTML tags are passed.", () => {
    instance.queueDeletions('Some <strong>HTML</strong>. And <i>more</i>.');
    expect(instance.queue).toHaveLength(21);
    expect(instance.queue).toMatchSnapshot();
  });
});
