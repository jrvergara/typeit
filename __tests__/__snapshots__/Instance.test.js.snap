// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queueDeletions() Queues correct length when HTML is passed. 1`] = `
Array [
  Array [
    [Function],
    250,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
]
`;

exports[`queueDeletions() Queues correct length when multiple HTML tags are passed. 1`] = `
Array [
  Array [
    [Function],
    250,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
]
`;

exports[`queueDeletions() Queues number when number is passed. 1`] = `
Array [
  Array [
    [Function],
    250,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
]
`;

exports[`queueDeletions() Queues string length when simple string is passed. 1`] = `
Array [
  Array [
    [Function],
    250,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
  Array [
    [Function],
    1,
  ],
]
`;
